include/seastar/core/sstring.hh

```diff
@@ -692,7 +692,13 @@ string_type to_sstring_sprintf(T value, const char* fmt) {
     char tmp[sizeof(value) * 3 + 2];
     auto len = std::sprintf(tmp, fmt, value);
     using ch_type = typename string_type::value_type;
+#pragma GCC diagnostic push
```

Comment:
clang claims to understand GCC's pragma with its additions. but it seems some older clang is not able to keep up with a more recent GCC. probably we have to conditionalize this on the clang version.

Reply:
I tested this on clang 14, so it's not old, but perhaps not new enough.
I found an even simpler workaround - add `#pragma GCC diagnostic ignored "-Wpragmas"`.
Which means, if the pragma worked, it worked. If it's not, then it's just ignored. In any case we care about the compilation working, not the pragram working :-)
I have a version I tested on both gcc and clang, I'll push it in a couple of minutes.